js/app/dashboard/redux/uiReducers.js

```diff
@@ -46,6 +59,45 @@ export default (state = initialState, action) => {
         ...state,
         loadingTourPanelsIds: _.xor([...state.loadingTourPanelsIds], [action.tourId])
       }
+    case APPEND_TO_UNASSIGNED_TASKS:
+      unassignedTasksIdsOrder = [...state.unassignedTasksIdsOrder]
+      _.remove(unassignedTasksIdsOrder, t => action.taskToRemoveIds.includes(t))
+      unassignedTasksIdsOrder = [...unassignedTasksIdsOrder, ...action.tasksToAppendIds]
+      return {
+        ...state,
+        unassignedTasksIdsOrder: unassignedTasksIdsOrder,
+      }
+
+    case INSERT_IN_UNASSIGNED_TASKS:
+      const tasksToInsertIds = action.tasksToInsert.map(t => t['@id'])
+      unassignedTasksIdsOrder = [...state.unassignedTasksIdsOrder]
+      _.remove(unassignedTasksIdsOrder, t => tasksToInsertIds.includes(t))
+      Array.prototype.splice.apply(unassignedTasksIdsOrder, Array.prototype.concat([ action.index, 0 ], tasksToInsertIds))
```

Comment:
Is not it more readable?

```
unassignedTasksIdsOrder.splice(action.index, 0, ...tasksToInsertIds)
```

Reply:
yes indeed, I was following the way it was written before (it was even more convoluted before, I couldnt understand the code ^^)

Reply:
i have simplified